packages/backend/src/apps/lettersg/actions/create-letter/index.ts

```diff
@@ -0,0 +1,125 @@
+import type { IRawAction } from '@plumber/types'
+
+import { ZodError } from 'zod'
+import { fromZodError } from 'zod-validation-error'
+
+import StepError, { GenericSolution } from '@/errors/step'
+
+import { requestSchema, responseSchema } from './schema'
+
+// TBC
+type LettersApiErrorData = {
+  message: string
+  success?: boolean
+  errors?: Record<string, string>[]
+}
```

Comment:
To check whether this is a union of success and error types, as they usually don't come together.

For example:

```typescript
type LettersApiErrorData = {
  message: string
  success: true
} | {
  success: false
  errors: Record<string, string>[]
}
```

Then TypeScript can perform type narrowing for you as long as you check the `success` flag

Reply:
this is a a specific error for only their validation of fields in the template, they will update us with a more specific error type, so I'm just gonna leave this out instead